pkg/agent/cache/manager/manager.go

```diff
@@ -0,0 +1,302 @@
+package manager
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/sirupsen/logrus"
+	spiffe_tls "github.com/spiffe/go-spiffe/tls"
+	"github.com/spiffe/go-spiffe/uri"
+	"github.com/spiffe/spire/pkg/agent/cache"
+	"github.com/spiffe/spire/pkg/common/util"
+	"github.com/spiffe/spire/proto/api/node"
+	proto "github.com/spiffe/spire/proto/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"io"
+	"sync"
+	"time"
+)
+
+type Manager interface {
+	UpdateCache(stop chan struct{})
+	fetchSVID(requests []EntryRequest, nodeClient node.NodeClient, wg *sync.WaitGroup)
+	getGRPCConn(svid []byte, key *ecdsa.PrivateKey) (*grpc.ClientConn, error)
+	expiredCacheEntryHandler(stop chan struct{})
+}
+
+type EntryRequest struct {
+	CSR   []byte
+	entry cache.CacheEntry
+}
+
+type manager struct {
+	managedCache     cache.Cache
+	serverCerts      []*x509.Certificate
+	serverAddr       string
+	serverSPIFFEID   string
+	errorCh          chan error
+	entryRequestCh   chan map[string][]EntryRequest
+	regEntriesCh     chan []*proto.RegistrationEntry
+	CacheEntryCh     chan cache.CacheEntry
+	spiffeIdEntryMap map[string]cache.CacheEntry
+	baseSVID         []byte
+	baseSVIDKey      *ecdsa.PrivateKey
+	baseSPIFFEID     string
+	regEntries       []*proto.RegistrationEntry
+	log              logrus.FieldLogger
+}
+
+func NewManager(managedCache cache.Cache,
+	serverCerts []*x509.Certificate,
+	serverSPIFFEID string,
+	serverAddr string,
+	errorCh chan error,
+	baseSVID []byte, baseSVIDKey *ecdsa.PrivateKey, regEntries []*proto.RegistrationEntry, logger logrus.FieldLogger) manager {
+	cert, err := x509.ParseCertificate(baseSVID)
+	if err != nil {
+		errorCh <- err
+	}
```

Comment:
Do you want to continue if there's an error?

Seems like just returning `(manager,error)` would be simpler.

Also, returning an un-exported struct (`manager`) from an exported function `NewManager` is kinda iffy.